python/cudf/cudf/core/dataframe.py

```diff
@@ -2522,6 +2522,13 @@ def _set_categories(col, cats):
 
         return df
 
+    def shift(self, periods=1, freq=None, axis=0, fill_value=None):
+        """Shift values in every column by `periods` positions and store the
+        output in a Dataframe.
+        """
+        assert axis in (None, 0) and freq is None and fill_value is None
+        return self._shift(periods)
```

Comment:
Can we just move this whole function to `frame.py`? I imagine we'd want the same restrictions on Series / Index / MultiIndex as of now.

Reply:
Do we want to let Pandas dictate the API of `Frame`? If so, that's fine, otherwise we need this layer as insulation.

Reply:
It's moreso that DataFrame, Series, MultiIndex, and Index all inherit from Frame, so we can define this once and reuse it across all 4 from `Frame`.

Reply:
So how about `.shift(...)` as the insulation layer (default args, etc), and `._shift()` for the underlying impl, leaving the door open to use `_shift()` elsewhere?